src/tests/rpp/test_defer.cpp

```diff
@@ -0,0 +1,62 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#include <snitch/snitch.hpp>
+
+#include <rpp/sources/defer.hpp>
+#include <rpp/sources/from.hpp>
+#include <rpp/sources/empty.hpp>
+#include <rpp/sources/error.hpp>
+#include <rpp/sources/never.hpp>
+
+#include "mock_observer.hpp"
+
+TEST_CASE("defer on different sources")
```

Comment:
Can we have test with mutable observable? For example, factory returns observable which emits incremented counter every subscribe. In case of direct usage such an observable we would see incremented counters while using defer - same initial value everytime

Reply:
done